src/sentry/models/groupforecast.py

```diff
@@ -0,0 +1,32 @@
+from django.db import models
+from django.utils import timezone
+
+from sentry.db.models import (
+    ArrayField,
+    BoundedPositiveIntegerField,
+    FlexibleForeignKey,
+    Model,
+    region_silo_only_model,
+)
+
+
+@region_silo_only_model
+class GroupForecast(Model):
+    """
+    Stores the forecast of expected counts of events for a Group
+
+    ``forecast`` will hold an array of integers. The length of the array
+    is the difference between `date_created` and `valid_until`. Each integer
+    maps to the days in the range of `date_created` and `valid_until`.
+
+    """
+
+    __include_in_export__ = False
+
+    group = FlexibleForeignKey("sentry.Group", unique=True)
```

Comment:
Will there definitely only ever be one row per group?

Reply:
Do different environments need separate forecasts?

Reply:
Ok, so I assume the write strategy here will be something like to use `create_or_update` for each forecast, and just overwrite the existing row? Or delete the old forecast and insert a new one in a transaction?

Reply:
Using `bulk_create` sounds good to me. The tricky part here is that since you have a unique constraint, you will need to delete/insert in a transaction to avoid a situation where the forecasts are missing for a period of time. So probably limit the batch size just to limit the amount of time the transaction is running.